tests/ssl-opt.sh

```diff
@@ -5622,7 +5632,7 @@ MAX_IM_CA='8'
 # are in place so that the semantics are consistent with the test description.
 requires_config_value_equals "MBEDTLS_X509_MAX_INTERMEDIATE_CA" $MAX_IM_CA
 requires_full_size_output_buffer
-requires_key_exchange_with_cert_in_tls12_or_tls13_enabled
+requires_key_exchange_with_cert_in_tls12_or_tls13_enabled ECDSA
```

Comment:
It's not cleat to me why this case and the following use `requires_key_exchange_with_cert_in_tls12_or_tls13_enabled ECDSA` while the rest of `max_int` tests use `requires_any_configs_enabled $TLS1_2_KEY_EXCHANGES_WITH_ECDSA_CERT`.

Also, I still feel like this could be auto-detected based on `dir-maxpath` the same way we do for `server5` and `server7`. It's the same reasoning: we load a certificate (chain) that we know uses ECDSA, and the requirements follow from that. Or am I missing something?

Reply:
> It's not cleat to me why this case and the following use requires_key_exchange_with_cert_in_tls12_or_tls13_enabled ECDSA while the rest of max_int tests use requires_any_configs_enabled $TLS1_2_KEY_EXCHANGES_WITH_ECDSA_CERT.

That's a good question indeed. I found this difference when I first started looking at this and I didn't pay much attention, just focusing on solving the issue. However your point makes sense to me, so I will align all the test cases

> Also, I still feel like this could be auto-detected based on dir-maxpath the same way we do for server5 and server7. [...] Or am I missing something?

Nope, you didn't miss anything; you're right. I will also fix this ;)

Reply:
> Also, I still feel like this could be auto-detected based on dir-maxpath the same way we do for server5 and server7. It's the same reasoning: we load a certificate (chain) that we know uses ECDSA, and the requirements follow from that. Or am I missing something?

I'm actually not a big fan of adding hidden dependencies. It increases the complexity of `ssl-opt.sh` tests, we have to follow not only the lines preceding each test, but also a set of arbitrary string matches. If someone adds a test that coincidentally matches one of the requirements, they might scratch their head because of why a given test isn't executed in a certain config combo...
 It's not a blocker to me, as I understand that it removes some repeated lines, but I wanted to voice my concern :)